### PR TITLE
hide replaced constructions when tech requirements differ

### PR DESCRIPTION
**Pull request by 9kgsofrice**
just a small change to the order of things, previously if you had unique constructions that used a different tech, the original still showed up in the tech tree

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/tech/Technology.kt [hunk 1]
```diff
@@ -82,11 +82,11 @@ class Technology {
 
     fun getEnabledBuildings(civInfo: CivilizationInfo): List<Building> {
         var enabledBuildings = civInfo.gameInfo.ruleSet.buildings.values.filter {
-            it.requiredTech == name &&
-                    (it.uniqueTo == null || it.uniqueTo == civInfo.civName)
+                    it.uniqueTo == null || it.uniqueTo == civInfo.civName
         }
         val replacedBuildings = enabledBuildings.mapNotNull { it.replaces }
         enabledBuildings = enabledBuildings.filter { it.name !in replacedBuildings }
+        enabledBuildings = enabledBuildings.filter { it.requiredTech == name }
```
Comment (yairm210) [inline core/src/com/unciv/models/ruleset/tech/Technology.kt:89]: It should be both - both not replaced AND having the same tech

**Comment by yairm210** [comment]
Actually now that I look at it, we can replace the whole 'find which buildings are replaced' with just going over all the buildings enabled by the tech and checking that `civInfo.getEquivalentBuilding(building.name)==building`

**Comment by logicminimal** [comment]
this pr is probably outdated by now

**Comment by yairm210** [comment]
Probably
@9kgsofrice You can always reopen it if you want :)

**Comment by logicminimal** [comment]
> Probably
> @9kgsofrice You can always reopen it if you want :)

why wasnt this merged anyway?

**Pull request closed without merging**